lib/Parse/ParsePattern.cpp

```diff
@@ -888,6 +879,56 @@ void Parser::parseAsyncThrows(
   }
 }
 
+/// Parse a type in parentheses after a `throws` or `rethrows`
+///
+ParserResult<TypeRepr> Parser::parseThrowsType() {
+  // look for the left paren for throws (_type_)
+  BacktrackingScope backtrack(*this);
```

Comment:
if I'm correct, we don't need backtracking here

Reply:
I'm not sure--I thought I had established that I needed it but I might have cleared that up. I will think about it again.